src/validation.rs

```diff
@@ -445,6 +465,45 @@ impl Validator {
             ))
         }
     }
+
+    fn validate_passed_call_parameter_size(
```

Comment:
I have the impression that the resolver already solves that problem for us. Should we be able to find downcasts by looking at the Type-Annotation -> TypeHint-Annotation for every statement?

Lets discuss this in person. Also see the unit test I uploaded to test my hypothesis

Reply:
I'm not 100% sure if the the special handling of pointers and by-ref parameters is already handled correctly by the resolver-approach